scripts/BilbyCustomFunctions_Reduction.py

```diff
@@ -0,0 +1,343 @@
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +from mantid.simpleapi import *
+
+import csv
+import math
+from itertools import product
+import sys
+from mantid.simpleapi import MoveInstrumentComponent, CropWorkspace
+
+
+#######################################################################################
+# REDUCTION ###########################################################################
+#######################################################################################
+def string_boolean(line):
+    """ Convert string to boolean; needed to read "true" and "false" from the csv Data reduction settings table """
+
+    if line == 'false':
+        bool_string = False
+    elif line == 'true':
+        bool_string = True
+    else:
+        print("Check value of {}".format(line))
+        print("It must be either True or False")
+        sys.exit()
+    return bool_string
+
+
+#######################################################################################
+
+def read_convert_to_float(array_strings):
+    """ Needed to convert binning parameters from the csv file into the float numbers """
+
+    array = [x.strip() for x in array_strings.split(',')]
+    array = [float(x) for x in array]
+    if (len(array) != 3):
+        print("Check input parameters; binning parameters shall be given in a format left_value, step, right_value.")
+        sys.exit()
+    return array
+
+
+###########################################################################################
+
+def FilesListReduce(filename):
+    """ Creat array of input reduction settings """
+    parameters = []
+    with open(filename) as csv_file:
+        reader = csv.DictReader(csv_file)
+        for row in reader:
+            if row['index'] == '':
+                continue
+            if row['index'] == 'END':
+                break
+            parameters.append(row)
+    return parameters
+
+
+#######################################################################################
+# ===============================================================
+# Function to extract list of lines in the csv file to be processed; input format is a combination of digits, '-' ,',' or empty space
+# returns list of numbers to be processed
+
+def evaluate_files_list(numbers):
+    """ Needed for FilesToReduce, see below """
+
+    expanded = []
+    for number in numbers.split(","):
+        if "-" in number:
+            start, end = number.split("-")
+            nrs = range(int(start), int(end) + 1)
+            expanded.extend(nrs)
+        else:
+            expanded.append(int(number))
+    return expanded
+
+    #######################################################################################
+
+
+def FilesToReduce(parameters, evaluate_files):
```

Comment:
Again shouldn't functions in python be in snake_case unless they need to be that due to inheriting from a C++ class or the like?